src/formatDate.js

```diff
@@ -50,7 +50,47 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateArr = date.split(fromFormat[3]);
+  const formatedArr = [];
+
+  for (let i = 0; i < toFormat.length - 1; i++) {
+    for (let j = 0; j < fromFormat.length - 1; j++) {
+      if (toFormat[i][0] === fromFormat[j][0]) {
+        if (toFormat[i][0] === 'Y') {
+          formatedArr[i] = convertYear(dateArr[j], toFormat[i].length);
+          break;
+        }
+
+        formatedArr[i] = dateArr[j];
+        break;
+      }
+    }
+  }
+
+  function convertYear(year, length) {
+    if (
+      (length === 4 && year.length === 4)
+      || (length === 2 && year.length === 2)
+    ) {
```

Comment:
```suggestion
    if (length === year.length) {
```
looks like it can be simplier